public/app/features/alerting/unified/components/receivers/form/fields/utils.ts

```diff
@@ -20,7 +20,7 @@ import { Template } from './TemplateSelector';
 export function parseTemplates(templatesString: string): Template[] {
   const templates: Record<string, Template> = {};
   const stack: Array<{ type: string; startIndex: number; name?: string }> = [];
-  const regex = /{{(-?\s*)(define|end|if|range|else|with|template)(\s*.*?)?(-?\s*)}}/gs;
+  const regex = /{{\s*(-?)\s*(define|end|if|range|else|with|template|block)\b(.*?)\s*(-?)\s*}}/gs;
```

Comment:
The start delimiter seems incorrect as that would match `{{ -` but the syntax only allows `{{` or `{{-`. Dito for the end delimiter.